### PR TITLE
Exposed error code and message

### LINKED ISSUE #84: Expose CURL error
When the curl_exec() fails, it returns `false`. The only info developers are left with is "Bad HTTP response".
For example, on an invalid SSL cert, I would only get "Bad HTTP response" error, which is a bit limited.

The error messages should be exposed using curl_error(), as it is done in PR #82

### PR DESCRIPTION
**Pull request by angelo983**
- Exposed error code and message in Exception response
- Added comments to curl params:
CURLOPT_SSL_VERIFYPEER
CURLOPT_CAINFO
CURLOPT_CAPATH
to remind ssl certificate chain checking.

| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Better debug informations showing real API response
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #84.
| How to test?      | Call an API endpoint with wrong or incomplete payload
| Possible impacts? | Exception and response managing

### CODE REVIEW

**Comment by matks** [comment]
Hi @angelo983  plenty of maintainers are enjoying some time off this week and the rest of us try to focus on the critical stuff 😅  so review might come a little late

**Comment by atomiix** [comment]
hi @angelo983, thanks for your contribution.
Would you mind filling the PR description table and maybe [create an issue](https://github.com/PrestaShop/PrestaShop-webservice-lib/issues/new) with a bit more detail about what your PR aims to do? It would be easier for us to understand and review your code as well as the QA team to check that everything works as expected.

Thanks

**Comment by matks** [comment]
Hello everyone, unfortunately there has been no news from you for more than 3 months, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale 😞 .

Feel free to ping me to reopen 😉 

Thanks!

**Comment by angelo983** [comment]
@matks PR description table just filled

**Comment by angelo983** [comment]
@matks I pinged you. I'm a little disappointed of how you value contributions. Guidelines says "may" and not "must".
I do not understand your concern to close, since I too had waited two months for a first feedback.

**Comment by matks** [comment]
> @matks I pinged you. I'm a little disappointed of how you value contributions. Guidelines say "may" and not "must" close pull requests after 30 days of inactivity. I do not understand your concern to close, since I too had waited two months for a first feedback.

@angelo983 I was on holidays when you pinged me and I just got back 😄 this is why I did not answer. I think you can understand people need some vacations sometimes 😄 .

Thank you for filling the description table and as I said i can now reopen 😉

**Review by matks — COMMENTED** [review]
I just reviewed the PR it is mostly ok, i suggested some improvements ;)

File: PSWebServiceLibrary.php [hunk 1]
```diff
@@ -137,6 +148,9 @@ protected function getCurlDefaultParams()
             CURLOPT_HTTPAUTH => CURLAUTH_BASIC,
             CURLOPT_USERPWD => $this->key . ':',
             CURLOPT_HTTPHEADER => array('Expect:'),
+			//CURLOPT_SSL_VERIFYPEER => false,
```
[the remark this answers is not in the document]
Comment (angelo983) [reply PSWebServiceLibrary.php:151]: Ok, I will add a comment

File: PSWebServiceLibrary.php [hunk 2]
```diff
@@ -92,36 +92,47 @@ function __construct($url, $key, $debug = true)
...
             default:
                 throw new PrestaShopWebserviceException(
-                    'This call to PrestaShop Web Services returned an unexpected HTTP status of:' . $status_code
+                    'This call to PrestaShop Web Services returned an unexpected HTTP status of:' . $request['status_code']
                 );
         }
+
+        if (!empty($error_message)) {
+            $response = $this->parseXML($request['response']);
+            $errors = $response->children()->children();
```
[the remark this answers is not in the document]
Comment (angelo983) [reply PSWebServiceLibrary.php:131]: If has no children $errors will be null, I added the check in if condition

File: PSWebServiceLibrary.php [hunk 3]
```diff
@@ -92,36 +92,47 @@ function __construct($url, $key, $debug = true)
      *
      * @throws PrestaShopWebserviceException if HTTP status code is not 200 or 201
      */
-    protected function checkStatusCode($status_code)
+    protected function checkStatusCode($request)
```
[the remark this answers is not in the document]
Comment (angelo983) [reply PSWebServiceLibrary.php:99]: Updated PhPDoc

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Hello @angelo983 thank you the PR is valid and I tested it

**Pull request merged**